storm-server/src/main/java/org/apache/storm/scheduler/resource/ResourceAwareScheduler.java

```diff
@@ -56,6 +59,10 @@
     private int schedulingTimeoutSeconds;
     private ExecutorService backgroundScheduling;
 
+    // record evicted topologies on each scheduling round, only used in test purpose now
+    private boolean forTest = false;
+    private Set<String> evictedTopologies = new HashSet<>();
+
```

Comment:
Can this be removed. Use TestUtilsForResourceAwareScheduler#assertTopologiesNotScheduled.

Reply:
Unfortunately no, I have to add this to check whether topo was evicted since in some scenarios, evicted topo would be put back in the same round of scheduling.